aim/web/ui/src/components/MediaSet/MediaSet.tsx

```diff
@@ -507,6 +383,166 @@ const MediaGroupedList = React.memo(function MediaGroupedList({
 },
 areEqual);
 
+function DepthSlider({
+  index,
+  pathValue,
+  depth,
+  onDepthChange,
+}: {
+  index: number;
+  pathValue: string | string[];
+  depth: number;
+  onDepthChange: (value: number, index: number) => void;
+}) {
+  return (
+    <ErrorBoundary>
+      <div
+        className='MediaSet__container__sliderContainer'
+        style={{ height: MEDIA_SET_SLIDER_HEIGHT }}
+      >
+        <Slider
+          aria-labelledby='track-false-slider'
+          track={false}
+          valueLabelDisplay='off'
+          getAriaValueText={(value) => `${pathValue[value]}`}
+          value={depth}
+          onChange={(e, value) => onDepthChange?.(value as number, index)}
+          step={null}
+          marks={(pathValue as string[]).map((l, i) => ({ value: i }))}
+          min={0}
+          max={pathValue.length - 1}
+          prevIconNode={
+            <Button
+              onClick={() => {
+                if (depth > 0) onDepthChange?.(depth - 1, index);
+              }}
+              className='prevIconBtn'
+              disabled={depth <= 0}
+              size='small'
+              withOnlyIcon
+            >
+              <Icon name='arrow-left' fontSize={10} />
+            </Button>
+          }
+          nextIconNode={
+            <Button
+              onClick={() => {
+                if (depth < pathValue.length - 1)
+                  onDepthChange?.(depth + 1, index);
+              }}
+              className='nextIconBtn'
+              disabled={depth >= pathValue.length - 1}
+              size='small'
+              withOnlyIcon
+            >
+              <Icon name='arrow-right' fontSize={10} />
+            </Button>
+          }
+        />
+      </div>
+    </ErrorBoundary>
+  );
+}
+
+function DepthDropdown({
```

Comment:
please move this component to a separate file

Reply:
done , thanks